src/runtime/execution.rs

```diff
@@ -234,19 +234,33 @@ impl ExecutionState {
     {
         Self::with(|state| {
             let task_id = TaskId(state.tasks.len());
-            let task = Task::from_future(future, stack_size, task_id, name);
+            let clock = state.increment_clock_mut(); // Increment the parent's clock
+            clock.extend(task_id); // and extend it with an entry for the new task
+            let task = Task::from_future(future, stack_size, task_id, name, clock.clone());
             state.tasks.push(task);
             task_id
         })
     }
 
-    pub(crate) fn spawn_thread<F>(f: F, stack_size: usize, name: Option<String>) -> TaskId
+    pub(crate) fn spawn_thread<F>(
+        f: F,
+        stack_size: usize,
+        name: Option<String>,
+        mut initial_clock: Option<VectorClock>,
+    ) -> TaskId
     where
         F: FnOnce() + Send + 'static,
     {
         Self::with(|state| {
             let task_id = TaskId(state.tasks.len());
-            let task = Task::from_closure(f, stack_size, task_id, name);
+            let clock = if let Some(ref mut clock) = initial_clock {
+                clock
+            } else {
+                // Inherit the clock of the parent thread (which spawned this task)
+                state.increment_clock_mut()
+            };
+            clock.extend(task_id); // and extend it with an entry for the new thread
```

Comment:
I think this is enough.  We update the parent's clock (incrementing its own entry and adding a 0-entry for the child), and this becomes the starting clock for the child.  The only causal knowledge the parent has about the child after a successful spawn is that the child started execution of its closure.  But the child's clock at that point is the same as the parent, and so the two are causally related.

Reply:
What I was trying to ask (and not explaining very well) was whether we wanted to update the child’s view of its own clock to some non-zero value when first spawned. But it doesn’t matter because you can’t compare the clocks of different threads, so starting a threads own clock at 0 is indistinguishable from starting it at some non-zero value.

Reply:
Yes, that was my thinking too.